### PR TITLE
added pysa vscode extension template

### PR DESCRIPTION
**Pull request by m0mosenpai**
Cloned Pyre's existing VSCode extension code and changed README, package.json for Pysa. This sets up a template for further development of extension and completes [Step #1](https://github.com/MLH-Fellowship/pyre-check/issues/2) in the overall roadmap to creating a Language Server extension for Pysa.

### CODE REVIEW

File: tools/ide_plugins/pysa-vscode/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		"target": "es6",
+		"outDir": "out",
+		"lib": [
+			"es6"
+		],
+		"sourceMap": true,
+		"rootDir": "src",
+		"strict": true   /* enable all strict type-checking options */
+		/* Additional Checks */
+		// "noImplicitReturns": true, /* Report error when not all code paths in function return a value. */
```
Comment (gbleaney) [inline tools/ide_plugins/pysa-vscode/tsconfig.json:13]: Why disable these? Looks like they're all enabled (plus the additional `noImplicitAny`) in the pyre plugin:
https://github.com/facebook/pyre-check/blob/master/tools/ide_plugins/vscode/tsconfig.json

**Review by gbleaney — CHANGES_REQUESTED** [review]

File: tools/ide_plugins/pysa-vscode/src/main.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		}
+	}
+}
+
+export async function activate(_: vscode.ExtensionContext) {
+
+    let debugOptions = { execArgv: ["--nolazy", "--inspect=6009"] };
+
+    let serverOptions = {
+	command: "pyre",
```
Comment (arthaud) [inline tools/ide_plugins/pysa-vscode/src/main.ts:39]: just adding on @gbleaney: here you are using a tab but the previous line is using spaces. Let's try to be consistent :)

File: tools/ide_plugins/pysa-vscode/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	export function dispose() {
+		if (configurationListener) {
+			configurationListener.dispose();
+		}
+	}
+}
+
+export async function activate(_: vscode.ExtensionContext) {
+
+    let debugOptions = { execArgv: ["--nolazy", "--inspect=6009"] };
```
[the remark this answers is not in the document]
Comment (m0mosenpai) [reply tools/ide_plugins/pysa-vscode/src/main.ts:36]: Yeah you're right. I might have been fiddling with some debugging stuff and forgot to remove this line. My bad

File: tools/ide_plugins/pysa-vscode/package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  "displayName": "pysa-vscode",
+  "description": "VSCode extension for Pysa",
+  "keywords": [
+    "python",
+    "pysa",
+    "typechecking"
+  ],
+  "main": "./out/main",
+  "activationEvents": [
+    "workspaceContains:.pyre_configuration"
```
[the remark this answers is not in the document]
Comment (m0mosenpai) [reply tools/ide_plugins/pysa-vscode/package.json:18]: Shall I add it anyways or leave it for later?

File: tools/ide_plugins/pysa-vscode/src/main.ts (hunk 2 above)
Comment (m0mosenpai) [reply tools/ide_plugins/pysa-vscode/src/main.ts:39]: I just realised VScode wasn't converting my tabs to 4 spaces. Fixed it.

**Comment by m0mosenpai** [comment]
Sorry for the mess caused by the space/tab thing and the auto generated files. I used the extension generator `yo code` for this folder and I tried removing things I knew we didn't need, but there were still others which I wasn't sure of so I decided to wait for feedback on this.

`.eslintrc.json` seems to be a JS/TS linter config file. The files in the `.vscode` directory are probably there so the vscode recognizes and runs the extension correctly. As for `tsconfig.json`, I don't know much about it other than that it is TS related.

File: tools/ide_plugins/pysa-vscode/package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (gbleaney) [reply tools/ide_plugins/pysa-vscode/package.json:20]: You can leave it for now. I just wanted to make you aware of it

File: tools/ide_plugins/pysa-vscode/src/main.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        if (configurationListener) {
+            configurationListener.dispose();
+        }
+    }
+}
+
+export async function activate(_: vscode.ExtensionContext) {
+
+    let serverOptions = {
+    command: "pyre",
```
Comment (gbleaney) [inline tools/ide_plugins/pysa-vscode/src/main.ts:37]: Indentation

File: tools/ide_plugins/pysa-vscode/src/main.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    let clientOptions: LanguageClientOptions = {
+    documentSelector: [{scheme: 'file', language: 'python'}],
+    synchronize: {
+        // Notify the server about file changes to '.clientrc files contain in the workspace
+        fileEvents: vscode.workspace.createFileSystemWatcher('**/.clientrc'),
+    }
+    };
+    
+    const languageClient = new LanguageClient(
+    'pyre',
```
Comment (gbleaney) [inline tools/ide_plugins/pysa-vscode/src/main.ts:50]: Indentation

**Review by gbleaney — CHANGES_REQUESTED** [review]
Looking pretty good. A few tweaks and we'll be ready to land!

**Comment by gbleaney** [comment]
Alright, landing!

**Pull request closed without merging**